### PR TITLE
IE11 FieldInline corrections

### PR DESCRIPTION
**Pull request by lukelooker**
### :sparkles: Changes

- IE11 Compatible grid layout
- Reposition "detail" on FieldInline to be more useful
- Unrelated IE11 fix for Avatar (use flex instead of grid definition)
- Replace enzmye-powered tests with react-testing-library
- Replace snapshots with stories

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [ ] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [ ] Checked for a11y impacts
- [x] Check for image-snapshot changes (run `yarn image-snapshots` locally)
- [x] PR is ideally < ~400LOC

### :camera: Screenshots

Screen Shot 2020-10-16 at 9 36 05 AM

### CODE REVIEW

File: packages/components/src/Form/Fields/FieldCheckbox/FieldCheckbox.story.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+export const ReadOnly = Template.bind({})
+ReadOnly.args = {
+  ...Basic.args,
+  readOnly: true,
+}
+
+export const Error = Template.bind({})
+Error.args = {
+  ...Basic.args,
+  readOnly: true,
```
Comment (dbchristopher) [inline packages/components/src/Form/Fields/FieldCheckbox/FieldCheckbox.story.tsx:81]: just double checking that `readOnly` is intentional here, and not a copy/paste artifact

**Review by dbchristopher — APPROVED** [review]
looks good! hooray for deleting snapshots 🎉

File: packages/components/src/Form/Fields/FieldCheckbox/FieldCheckbox.story.tsx (hunk 1 above)
Comment (lukelooker) [inline packages/components/src/Form/Fields/FieldCheckbox/FieldCheckbox.story.tsx:81]:
```suggestion
```

File: packages/components/src/Form/Fields/FieldCheckbox/FieldCheckbox.story.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  validationMessage: { message: 'This is an error', type: 'error' },
+}
+
+export const DetailError = Template.bind({})
+DetailError.args = {
+  ...Detail.args,
+  readOnly: true,
```
Comment (lukelooker) [inline packages/components/src/Form/Fields/FieldCheckbox/FieldCheckbox.story.tsx:88]:
```suggestion
```

**Review by dbchristopher — APPROVED** [review]
Looks good!

**Pull request merged**